compiler/Makefile

```diff
@@ -158,7 +158,7 @@ $(CHPL_CONFIG_CHECK): | $(CHPL_BIN_DIR)
 	echo $(CHPL_MAKE_COMPILER_SUBDIR) > $(CHPL_CONFIG_CHECK)
 
 $(CHPL): $(CHPL_OBJS) $(CHPL_CONFIG_CHECK) | $(CHPL_BIN_DIR)
-	$(CXX) $(LDFLAGS) -o $@ $(CHPL_OBJS) $(LIBS) $(LLVM_LIBS)
+	$(CXX) $(LDFLAGS) -o $@ $(CHPL_OBJS) $(CHPL_MAKE_THIRD_PARTY_HOST_LINK_ARGS) $(LIBS) $(LLVM_LIBS)
```

Comment:
Not needed for this PR, but I wonder if we should fold `LLVM_LIBS` into `CHPL_MAKE_THIRD_PARTY_HOST_LINK_ARGS`